src/components/molecules/tabs/request/response/HeadersDisplayTable.tsx

```diff
@@ -0,0 +1,59 @@
+import { AccordionGroup, Accordion, AccordionSummary, AccordionDetails, Table } from '@mui/joy';
+import { SPHeaders } from '../../../../../types/application-data/application-data';
+
+export function HeadersDisplayTable({
+	headers,
+	label,
+}: {
+	headers: Record<string, string> | SPHeaders;
+	label: 'request' | 'response';
+}) {
+	let display = <></>;
+	if (headers.__data && typeof headers.__data != 'string') {
+		if (headers.__data.length == 0) {
+			return <></>;
```

Comment:
These nested returns are hell to follow, I think? Why not have display = null (with a type compatible with <></> as well) and then get rid of these returns and only have them fill display if it actually has content. Then you can just do a single check if display==null return <></>?